packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -718,6 +749,44 @@ function commitRoot(root: FiberRoot, finishedWork: Fiber): void {
     legacyErrorBoundariesThatAlreadyFailed = null;
   }
   onCommit(root, earliestRemainingTimeAfterCommit);
+
+  if (enableInteractionTracking) {
+    __interactionsRef.current = prevInteractions;
+
+    let caughtError = null;
+    let subscriber;
+
+    try {
+      subscriber = __subscriberRef.current;
+      if (subscriber !== null && root.memoizedInteractions.size > 0) {
+        // Interaction threads are unique per root and expiration time.
+        // DO NOT CHANGE THIS VALUE without also updating other threadID values.
+        const threadID =
+          committedExpirationTime * 1000 + root.interactionThreadID;
+        subscriber.onWorkStopped(root.memoizedInteractions, threadID);
+      }
+    } catch (error) {
+      caughtError = error;
+    } finally {
+      // Now that we're done, check the completed batch of interactions.
+      // If no more work is outstanding for a given interaction,
+      // We need to notify the subscribers that it's finished.
+      committedInteractions.forEach(interaction => {
+        interaction.__count--;
+        if (subscriber !== null && interaction.__count === 0) {
+          try {
+            subscriber.onInteractionScheduledWorkCompleted(interaction);
+          } catch (error) {
+            caughtError = caughtError || error;
+          }
+        }
+      });
+
+      if (caughtError) {
+        throw caughtError;
```

Comment:
`commitRoot` should never throw. There could be additional work at lower priority, or on another root. This is probably why you had to use `try/catch` instead of `try/finally` in the test renderer.

If interaction subscribers throw an error, we should only rethrow them once React is completely finished with its other work. So at the end of `performWork`.

Reply:
To be clear, `try`/`finally` doesn't make sense at all in the test renderer case. I only added the `try`/`catch` to recover from a corrupted state.

I don't _think_ this is related. In that, this code path isn't even executed in the tests that fail without the test renderer fix (e.g. Profiler › interaction tracking › error handling)

Reply:
On re-reading this code, I think the `caughtError` approach was silly and I should have just made use fo the existing `hasUnhandledError` / `unhandledError` variables 😄